### ISSUE TITLE
IOS build failed on adding flutter_downloader

### ISSUE THREAD

**Issue by sagardspeed2** [fluttercommunity/flutter_downloader#519]
<img width="1680" alt="Screenshot 2021-09-02 at 12 52 00 AM" src="https://user-images.githubusercontent.com/31651612/131733152-fb1fafdf-929c-4fcc-afe8-06f020cca8fb.png">

I am getting this error when trying to build IOS,
but the android build works fine.

here is my appdelegate

```
import UIKit
import Flutter
import flutter_downloader
import Firebase

@UIApplicationMain
@objc class AppDelegate: FlutterAppDelegate {
  override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {
        FirebaseApp.configure()
           if(!UserDefaults.standard.bool(forKey: "Notification")) {
          UIApplication.shared.cancelAllLocalNotifications()
           UserDefaults.standard.set(true, forKey: "Notification")
       }

       if #available(iOS 10.0, *) {
         UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
       }
    GeneratedPluginRegistrant.register(with: self)
    FlutterDownloaderPlugin.setPluginRegistrantCallback(registerPlugins)
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }
}

private func registerPlugins(registry: FlutterPluginRegistry) { 
    if (!registry.hasPlugin("FlutterDownloaderPlugin")) {
       FlutterDownloaderPlugin.register(with: registry.registrar(forPlugin: "FlutterDownloaderPlugin")!)
    }
}
```

**Comment by aravind-b-dev** [comment]
Please follow the proper documentaion
https://pub.dev/packages/flutter_downloader

I faced the same problem

**Comment by aravind-b-dev** [comment]
Here my appdelagate file


    import UIKit
    import Flutter
    import Firebase
    import FirebaseMessaging
    import UserNotifications
    import flutter_downloader

    @UIApplicationMain
    @objc class AppDelegate: FlutterAppDelegate {
    var firebaseToken : String = "";
    override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
      ) -> Bool {
      Messaging.messaging().delegate = self;
      FirebaseApp.configure()
    GeneratedPluginRegistrant.register(with: self)
    FlutterDownloaderPlugin.setPluginRegistrantCallback(registerPlugins)
    Messaging.messaging().isAutoInitEnabled = true;
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
      }

      override func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken : Data){
          print("X__APNS: \(String(describing: deviceToken))")
         Messaging.messaging().apnsToken = deviceToken;
      //Messaging.messaging().setAPNSToken(deviceToken, type:MessagingAPNSTokenType.prod )
      }

      override func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
          print("X_ERR",error);
      }

       func registerForFirebaseNotification(application : UIApplication){
      //    Messaging.messaging().delegate     = self;
          if #available(iOS 10.0, *) {
              //UNUserNotificationCenter.current().delegate = self ;
              UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
              let authOpt : UNAuthorizationOptions = [.alert, .badge, .sound];
              UNUserNotificationCenter.current().requestAuthorization(options: authOpt, completionHandler: {_, _ in})
              UNUserNotificationCenter.current().delegate = self ;
          }else{
              let settings : UIUserNotificationSettings = UIUserNotificationSettings(types: [.alert, .badge, .sound], categories: nil)
              application.registerUserNotificationSettings(settings);
          }
          application.registerForRemoteNotifications();
      }
     }

      extension AppDelegate : MessagingDelegate{
      func messaging(_messagin : Messaging, didRecieveRegistrationToken fcmToken : String){
          self.firebaseToken = fcmToken;
          print("X__FCM: \(String(describing: fcmToken))")
      }
      func messaging(_messagin : Messaging, didRecieve remoteMessage : Any){
          //self.firebaseToken = fcmToken;
          print("REMOTE__MSG: \(remoteMessage)")
      }
      func application(_ application : UIApplication,didRecieveRemoteNotification uinfo: [AnyHashable : Any], fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void){
          print("WITH__APN: \(uinfo)")
      }
    }

    private func registerPlugins(registry: FlutterPluginRegistry) {
    if (!registry.hasPlugin("FlutterDownloaderPlugin")) {
        FlutterDownloaderPlugin.register(with: registry.registrar(forPlugin: "FlutterDownloaderPlugin")!)
        }
    }
